addons/mail/static/src/components/thread_needaction_preview/thread_needaction_preview_tests.js

```diff
@@ -92,6 +92,12 @@ QUnit.test('click on preview should mark as read and open the thread', async fun
     await this.start({
         hasChatWindow: true,
         async mockRPC(route, args) {
+            if (route.includes('channel_fold')) {
+                // [technical] This assertion should not happen when opening
+                // non-channels in chat window, because there is no server sync
+                // of fold state for them.
+                assert.step('channel_fold');
+            }
```

Comment:
Would have preferred making a technical test for that, I don't like polluting functional test with that.
Also instead of delaying crash with a `step`/non-matching `verifySteps`, just throw error ASAP.

Reply:
done

Reply:
But not using assert means the test is going to pass, and relying on runbot to crash on `console.error`

Reply:
I'm going to add it back because it's really confusing.